### PR TITLE
Use a .patch to comment out Slurm settings which need to be overwritten by OpenHPC

### PR DESCRIPTION
**Pull request by martin-g**
Related-to: https://github.com/openhpc/ohpc/pull/1830

### CODE REVIEW

**Comment by adrianreber** [comment]
How do you feel about having the changes from all the `echo` lines also be included in the patch?

**Comment by martin-g** [comment]
It will work! 
I just wonder how often we will need to adapt the patch when upgrading Slurm.
Looking at the Git history of https://github.com/SchedMD/slurm/commits/master/etc/slurm.conf.example I see that they make 1 or 2 changes per year. But most of the changes are in settings' values, not adding/removing settings (and thus change the line numbers). So probably it should be OK!

**Comment by adrianreber** [comment]
Right, having a patch is more work if upstream changes things. On the other hand it is probably more reliable to actually change the file as desired. `grep`, `sed`, `echo` will probably not give us feedback if something failed.

**Comment by martin-g** [comment]
OK! I will move those lines to the patch!

File: components/rms/slurm/SPECS/slurm.spec [hunk 1]
```diff
@@ -487,21 +490,7 @@ install -D -m755 contribs/sjstat %{buildroot}/%{_bindir}/sjstat
...
-echo "PartitionName=normal Nodes=c[1-4] Default=YES MaxTime=24:00:00 State=UP Oversubscribe=EXCLUSIVE" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf.ohpc
-# 5/5/2020 karl@oden.utexas.edu - enable configless option
-echo "SlurmctldParameters=enable_configless" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf.ohpc
-# 11/9/2021 karl@oden.utexas.edu - setup interactive jobs for salloc
-LaunchParameters=use_interactive_step >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf.ohpc
-# 6/3/16 nirmalasrjn@gmail.com - Adding ReturnToService Directive to starting config file (note removal of variable during above creation)
-echo "ReturnToService=1" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf.ohpc
+cp $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf.example $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf.ohpc
 # 9/17/14 karl.w.schulz@intel.com - Add option to drop VM cache during epilog
 sed -i '/^# No other SLURM jobs,/i \\n# Drop clean caches (OpenHPC)\necho 3 > /proc/sys/vm/drop_caches\n\n#' $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.epilog.clean
```
Comment (martin-g) [inline components/rms/slurm/SPECS/slurm.spec]: Here is an example of a silent error!
There is no `slurm.epilog.clean` at https://github.com/SchedMD/slurm/tree/slurm-22-05-10-1/etc, so this `sed` command does nothing!

Comment (martin-g) [reply components/rms/slurm/SPECS/slurm.spec]: The file has been removed 5 years ago with https://github.com/SchedMD/slurm/commits/master/etc/slurm.epilog.clean

Comment (adrianreber) [reply components/rms/slurm/SPECS/slurm.spec]: It still seems to exist:

```
[root@c1 ~]# rpm -qf /etc/slurm/slurm.epilog.clean
slurm-example-configs-ohpc-22.05.10-2.1.ohpc.2.6.2.x86_64
```

Comment (adrianreber) [reply components/rms/slurm/SPECS/slurm.spec]: https://github.com/openhpc/ohpc/blob/15dab81c0d831e768c50d9e5e1e7d98ea202057b/components/rms/slurm/SPECS/slurm.spec#L485

Comment (martin-g) [reply components/rms/slurm/SPECS/slurm.spec]: Yep!
I see now that it is copied in SOURCES/ - https://github.com/openhpc/ohpc/blob/3.x/components/rms/slurm/SOURCES/slurm.epilog.clean
All is good!
Except that we could move https://github.com/openhpc/ohpc/blob/15dab81c0d831e768c50d9e5e1e7d98ea202057b/components/rms/slurm/SPECS/slurm.spec#L506 directly in the file.

Comment (adrianreber) [reply components/rms/slurm/SPECS/slurm.spec]: Right. No need to modify the file with sed.

File: components/rms/slurm/SOURCES/slurm.conf.example.patch [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ # COMPUTE NODES
+-NodeName=linux[1-32] CPUs=1 State=UNKNOWN
+-PartitionName=debug Nodes=ALL Default=YES MaxTime=INFINITE State=UP
++#NodeName=linux[1-32] CPUs=1 State=UNKNOWN
++#PartitionName=debug Nodes=ALL Default=YES MaxTime=INFINITE State=UP
++
++
++
++# OpenHPC default configuration
++# 10/2/18 brad.geltz@intel.com - Enabling the task/affinity plugin to add the --cpu-bind option to srun for GEOPM
```
Comment (adrianreber) [inline components/rms/slurm/SOURCES/slurm.conf.example.patch]: Let's remove the dates and the email addresses from the file. We have the authors of the changes in git.

File: components/rms/slurm/SOURCES/slurm.epilog.clean [hunk 3]
```diff
@@ -30,6 +30,10 @@ do
 	fi
 done
 
+# 9/17/14 karl.w.schulz@intel.com - Add option to drop VM cache during epilog
+# Drop clean caches (OpenHPC)
+echo 3 > /proc/sys/vm/drop_caches
```
Comment (adrianreber) [inline components/rms/slurm/SOURCES/slurm.epilog.clean]: Let's remove the changes to the epilog file.

Comment (martin-g) [reply components/rms/slurm/SOURCES/slurm.epilog.clean]: Done!

File: components/rms/slurm/SOURCES/slurm.conf.example.patch (hunk 2 above)
Comment (martin-g) [reply components/rms/slurm/SOURCES/slurm.conf.example.patch]: Done!

**Comment by adrianreber** [comment]
Thanks

**Pull request merged**